tutor/src/components/terms-modal.jsx

```diff
@@ -1,34 +1,45 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import { observer } from 'mobx-react';
 import { action, computed } from 'mobx';
+import { observer, inject } from 'mobx-react';
 import { Modal, Button } from 'react-bootstrap';
 import classnames from 'classnames';
 import Branding from './branding/course';
 import User from '../models/user';
 import Course from '../models/course';
 import { map } from 'lodash';
 import String from '../helpers/string';
+import ModalManager from './modal-manager';
 
 export default
+@inject('modalManager')
 @observer
 class TermsModal extends React.Component {
 
   static propTypes = {
     canBeDisplayed: PropTypes.bool,
+    modalManager: PropTypes.instanceOf(ModalManager).isRequired,
   }
 
   @computed get title() {
     return String.toSentence(map(User.unsignedTerms, 'title'));
   }
 
-  @action.bound onAgreement() {
-    User.terms.sign();
+  @action.bound onAgreement() { User.terms.sign(); }
+
+  constructor(props) {
+    super(props);
+    this.priority = 10;
+    this.props.modalManager.queue(this);
+  }
+
```

Comment:
you could set `priority` as a instance variable outside the constructor.  Also how important is it that `modalManager.queue` be called before render?

Doing stuff in the constructor is a bit frowned on because React keeps saying that in future versions, it's possible that it might create a component, but then never render it and just dispose of it.  It's better to use `componentDidMount` and do work there, but the disadvantage of that method is that it's called after the first render.

Reply:
I'll try moving the queue to DidMount

Reply:
Moved them all to componentWillMount for now...